AllASPFuncs.py

```diff
@@ -1292,3 +1307,34 @@ def banner(text):
     length = len(text) + 8
     message = "\n" + "#" * length +'\n'+'#   ' + text + '   # \n'+ "#" * length
     Lager.debug(message)
+
+
+def prep_data_for_fit(images, err, sn_matrix, wgt_matrix):
+    '''
+    This function takes the data from the images and puts it into the form such
+    that we can analytically solve for the best fit using linear algebra.
+    '''
+    size = int(np.sqrt((images[0].size)))
+    tot_num = len(images)
+    det_num = len(sn_matrix)
+
+    # Flatten into 1D arrays
+    images = np.concatenate([arr.flatten() for arr in images])
+    err = np.concatenate([arr.flatten() for arr in err])
+
+    psf_zeros = np.zeros((np.size(images), tot_num))
+
+    for i in range(det_num):
```

Comment:
Define `tot_num - det_num + i` as a variable, and then use that variable in the indexing.
This is better because you're not re-typing an operation (I had to double-check that `(tot_num - det_num + i)` and `(tot_num - det_num) + i` were the same.) and because it will actually make the indexing shorter and easier to read.